### PR TITLE
Handling suppressed and temp basals

### PR DESCRIPTION
**Pull request by gniezen**
**WIP**: Do not merge.

### CODE REVIEW

**Comment by gniezen** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;DWsSEns0&#x2F;30-5-uploader-tandem-dotted-line-suppressed-info-for-temp-basals-and-suspends"> &lt;5&gt; [uploader] [tandem] &quot;dotted line&quot;&#x2F;suppressed info for temp basals and suspends</a>

**Comment by gniezen** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;DWsSEns0&#x2F;30-5-uploader-tandem-dotted-line-suppressed-info-for-temp-basals-and-suspends"> &lt;5&gt; [uploader] [tandem] &quot;dotted line&quot;&#x2F;suppressed info for temp basals and suspends</a>

File: lib/tandem/tandemSimulator.js [hunk 1]
```diff
@@ -79,6 +84,13 @@ exports.make = function(config){
     basal: function(event){
       ensureTimestamp(event);
       if (currBasal != null) {
+        if(currBasal.deliveryType === 'temp') {
+          if (currTempBasal !== null) {
+            currBasal.percent = currTempBasal.percent;
```
Comment (jebeck) [inline lib/tandem/tandemSimulator.js]: The percents aren't coming through anywhere. Check through all the steps to get them through? There's also an issue with the payloads - the issue that came up before with using the object builder's `set` method like a `with_*` method.

Comment (gniezen) [inline lib/tandem/tandemSimulator.js]: @jebeck Are you not seeing them anywhere? They're coming through for me:

screen shot 2015-11-30 at 11 16 10

Comment (jebeck) [inline lib/tandem/tandemSimulator.js]: Check your use(s) of truthiness. All your examples coming through are non-zero percentages.

**Comment by gniezen** [comment]
@jebeck I've made some more changes. Will you be able to test to see if they're now coming through. I'm also running a temp basal over midnight tonight to test the new-day issue, but as you have that already you'll be able to confirm if it's fixed.

File: lib/drivers/tandemDriver.js [hunk 2]
```diff
@@ -1595,6 +1595,14 @@ module.exports = function (config) {
           debug('Event with unhandled change type:', event);
           throw new Error('Unhandled combination of basal change types: ' + event.change_types.join('|'));
       }
+      if(postbasal.deliveryType === 'temp') {
```
Comment (jebeck) [inline lib/drivers/tandemDriver.js]: You can't have this logic above the `if (postbasal != null)` - errored on my pump.

Comment (gniezen) [inline lib/drivers/tandemDriver.js]: Ah, it looks like that happens when there's a pump shutdown event. I've moved the logic inside, thanks.

**Comment by gniezen** [comment]
Pinging @jebeck for (hopefully) last review and test cycle.

**Comment by jebeck** [comment]
Functionality LGTM now. I think the Tandem simulator tests just need some additions. Off the top of my head:
- testing for how simulating tempBasal works & helps to provide `suppressed` info (interaction with new day and with rate changes when the temp basal crosses multiple scheduled segments)
- final temp basals (esp. what you're doing with `time_left`)

**Comment by gniezen** [comment]
Merging; will create a separate card for writing more comprehensive Tandem-v2 unit tests.

**Pull request merged**